apps/detect/src/components/dashboard/header.tsx

```diff
@@ -9,13 +12,56 @@ export default function Header({
   count: React.ReactNode;
   right?: React.ReactNode;
 }) {
+  const { data: reportData } = useQuery(queries.detect.activity.logs());
+
+  const downloadReport = useCallback(async () => {
+    if (reportData) {
```

Comment:
nit: prefer early returns/guards

```suggestion
    if (!reportData) { return }
```